src/django_idom/hooks.py

```diff
@@ -30,3 +40,145 @@ def use_websocket() -> IdomWebsocket:
     if websocket is None:
         raise RuntimeError("No websocket. Are you running with a Django server?")
     return websocket
+
+
+_REFETCH_CALLBACKS: DefaultDict[FunctionType, set[Callable[[], None]]] = DefaultDict(
+    set
+)
+
+
+_Data = TypeVar("_Data")
+_Params = ParamSpec("_Params")
+
+
+def use_query(
+    query: Callable[_Params, _Data],
+    *args: _Params.args,
+    fetch_deferred_fields: bool = True,
+    **kwargs: _Params.kwargs,
+) -> Query[_Data]:
+    given_query = query
+    query, _ = use_state(given_query)
+    if given_query is not query:
+        raise ValueError(f"Query function changed from {query} to {given_query}.")
+
+    data, set_data = use_state(UNDEFINED)
+    loading, set_loading = use_state(True)
+    error, set_error = use_state(None)
+
+    @use_callback
```

Comment:
Is there a functional reason for having this be `use_callback`?

Reply:
In the context of the hook no, but if we don't the identity of the function will change on every render. Then, If this callback is a dependency in the user's code (e.g. an effect) we don't want to cause unnecessary renders.

This is common in hook libraries. For example, the `set_state` callback from the `use_state` hook does not change identity after a render. In application code, `use_callback` shows up less commonly and is just a way to improve performance where necessary since each usage has a memory cost.